### ISSUE TITLE
Support XSimd

### ISSUE THREAD

**Issue by anutosh491** [lcompilers/lc#86]
The project looks great. It would be nice if there could be some support for libraries providing C++ wrappers for SIMD intrinsics. 
A prime example for this could be 
1) https://github.com/xtensor-stack/xsimd
2) https://xsimd.readthedocs.io/en/latest/index.html

**Comment by certik** [comment]
Yes, for xsimd we should use our "SIMD" physical array in LCompilers, to generate vectorized code.
